lib/src/ast/expression.dart

```diff
@@ -2,21 +2,41 @@
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
 
-// TODO: This is a placeholder.
-//
-// Probably want to either:
-// - Port the original one
-// - Write a new one
-// - Use the Dart analyzer and special case pipes
-// - Use the Dart analyzer and transform pipes
+import 'package:analyzer/analyzer.dart';
+import 'package:angular_ast/src/expression/parser.dart';
+
+/// Wraps a parsed Dart [Expression] as an Angular [ExpressionAst].
 class ExpressionAst {
-  final String expression;
+  /// Dart expression.
+  final Expression expression;
 
+  /// Create a new expression AST wrapping a Dart expression.
   const ExpressionAst(this.expression);
 
+  /// Create a new expression AST by parsing [expression].
+  factory ExpressionAst.parse(
+    String expression, {
+    bool deSugarPipes: true,
+    String sourceUrl: '/test.dart',
+  }) {
+    // Fall back in case this comes from a test w/o a real source URL.
+    if (!sourceUrl.endsWith('.dart')) {
+      sourceUrl = '/test.dart';
+    }
+    return new ExpressionAst(parseExpression(
+      expression,
+      deSugarPipes: deSugarPipes,
+      sourceUrl: sourceUrl,
+    ));
+  }
+
   @override
-  bool operator ==(Object o) =>
-      o is ExpressionAst && o.expression == expression;
+  bool operator ==(Object o) {
+    if (o is ExpressionAst) {
+      return o.expression.toSource() == expression.toSource();
+    }
+    return false;
+  }
```

Comment:
Could be useful to add to the visitor as well (for instance, we'll probably put a scope of local variables on each of these, and then run a visitor that just resolves every Expression. Might not want to have to implement visitMustache, visitInput, etc etc just to get at all the Exrpessions).